openapi_python_client/parser/properties.py

```diff
@@ -575,3 +617,27 @@ def property_from_data(
         return _property_from_data(name=name, required=required, data=data)
     except ValidationError:
         return PropertyError(detail="Failed to validate default value", data=data)
+
+
+def type_string(no_optional: bool, prop: Property, inner_type_string: str, is_outer_union: bool = False) -> str:
+    if no_optional or (prop.required and not prop.nullable):
+        if is_outer_union:
+            return f"Union[{inner_type_string}]"
+        return inner_type_string
+    elif prop.nullable and prop.required:
+        if is_outer_union:
+            return f"Optional[Union[{inner_type_string}]]"
+        return f"Optional[{inner_type_string}]"
+    elif not prop.nullable and not prop.required:
+        return f"Union[Unset, None, {inner_type_string}]"
+    return f"Union[Unset, {inner_type_string}]"
+
+
+def query_method_arg(no_optional: bool, prop: Property, inner_type_string: str, is_outer_union: bool = False) -> str:
+    if no_optional or (prop.required and not prop.nullable):
```

Comment:
Potentially slightly more readable as:
```
type_string = inner_type_string
if is_outer_union:
	type_string = f"Union[{type_string}]"
if not no_optional and (not prop.required or prop.nullable):
	type_string = f"Optional[{type_string}]"
return type_string
```
Similarly for `type_string` above. Don't feel too strongly though, will leave it to you.